### PR TITLE
Add a command to run a query in 'dry run' mode

### PR DESCRIPTION
**Pull request by dan-nom**
Show success or failure messages through the `showInformationMessage` toasts.

### CODE REVIEW

**Comment by dan-nom** [comment]
I signed it!

File: src/extension.ts [hunk 1]
```diff
@@ -74,10 +79,16 @@ function query(queryText: string): Promise<any> {
...
       maximumBytesBilled: config.get("maximumBytesBilled"),
-      useLegacySql: config.get("useLegacySql")
+      useLegacySql: config.get("useLegacySql"),
+      dryRun: !!isDryRun
     })
     .then(data => {
       let job = data[0];
+      if (isDryRun) {
+        vscode.window.showInformationMessage(
+            `BigQuery dry run successful. Job ID: ${job.id}`);
```
Comment (elithrar) [inline src/extension.ts]: Might be worth re-using the template string below for consistency - and just appending "(dry run)" in this case.

File: src/extension.ts [hunk 2]
```diff
@@ -192,4 +203,13 @@ function runSelectedAsQuery(): void {
   }
 }
 
+function dryRun(): void {
```
Comment (elithrar) [inline src/extension.ts:215]: 👍

File: package.json [hunk 3]
```diff
@@ -29,6 +30,10 @@
       {
         "command": "extension.runSelectedAsQuery",
         "title": "BigQuery: Run selected text as Query"
+      },
+      {
+        "command": "extension.dryRun",
+        "title": "BigQuery: Dry run Query"
```
Comment (elithrar) [inline package.json]: "Run as Dry Run"

File: src/extension.ts [hunk 4]
```diff
@@ -192,4 +203,13 @@ function runSelectedAsQuery(): void {
...
+  try {
+    let queryText = getQueryText(vscode.window.activeTextEditor);
+    query(queryText, true);
```
Comment (elithrar) [inline src/extension.ts:218]: `query` calls `writeResults` - but that function only writes the result table out, not the job statistics. For a dry run, you very likely want to see bytes scanned, to determine whether your query was not only syntactically correct, but also hit the right partitions/clusters/isn't going to bill you too much.

**Review by elithrar — CHANGES_REQUESTED** [review]
Review feedback on `dryRun` job output.

File: src/extension.ts (hunk 1 above)
Comment (dan-nom) [reply src/extension.ts]: Done, thanks for the suggestion!

File: package.json (hunk 3 above)
Comment (dan-nom) [reply package.json]: Done (I played with the wording here a bit - nothing seems great!)

File: src/extension.ts (hunk 4 above)
Comment (dan-nom) [reply src/extension.ts:218]: Done, added `function writeDryRunSummary(jobId: string, numBytesProcessed: string)`

**Comment by dan-nom** [comment]
Thanks for the prompt review. PTAL.

**Comment by elithrar** [comment]
Merged - thanks @dan-nom!

**Pull request merged**